### ISSUE TITLE
Overriding nested (table) config vars clobbers defaults

### ISSUE THREAD

**Issue by chetan** [habitat-sh/habitat#1184]
Configuration is only shallow-merged[1] currently, which creates issues when overriding vars inside a table. For example, given this default config: 

```toml
[server]
port = "8080"
shutdown-port = "8005"
redirect-port = "8443"
```
(from https://github.com/habitat-sh/core-plans/blob/master/tomcat8/default.toml)

If you override the port using `HAB_TOMCAT="server = { port=9090 }"` then the other `[server]` configs like `shutdown-port` get clobbered in the final  config (entirely missing), resulting in empty strings when using those vars. 

[1] https://github.com/habitat-sh/habitat/blob/master/components/sup/src/service_config.rs#L313

**Comment by metadave** [comment]
potential fix here: https://github.com/habitat-sh/habitat/pull/1185

**Comment by juliandunn** [comment]
This got fixed (for realzies this time) in #1307 and #1308, released with Habitat 0.10.2.
